src/main/java/net/minecraftforge/event/entity/TeamEvent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.entity;
+
+import net.minecraft.scoreboard.Team;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+public class TeamEvent extends Event
+{
+    private final Team team;
+
+    public TeamEvent(Team team)
+    {
+        this.team = team;
+    }
+
+    public Team getTeam()
+    {
+        return this.team;
+    }
+
+    public static class CreateTeamEvent extends TeamEvent
+    {
+        public CreateTeamEvent(Team team)
+        {
+            super(team);
+        }
+    }
+
+    public static class RemoveTeamEvent extends TeamEvent
+    {
+        public RemoveTeamEvent(Team team)
+        {
+            super(team);
+        }
+    }
+
+    public static class EntityTeamEvent extends TeamEvent
+    {
+        private final String entityName;
+
+        public EntityTeamEvent(Team team, String entityName)
+        {
+            super(team);
+            this.entityName = entityName;
+        }
+
+        public String getEntityName()
```

Comment:
If it's a Player, it appears to be the GameProfile name, else it's UUID. It's definitely not a display name. The String is what is passed around for adding Entities to Teams. While I'm not sure if this is the best name (It's called about 5 different things throughout the code: player, playerName, username, name), a TextComponent definitely isn't the answer. I tried passing the actual Entity that was being handled around (As it's grabbed in the code before being dropped later in the code), it began to require way more lines to be changed to get that working.

Reply:
Try to write a javadoc for this method. Your explanation still confuses me. Do you mean that if a player's name part in gameprofile is not found, this will be uuid toString result? How about the entities, are these entities custom name or their unique/persistent id?